lake/writer.go

```diff
@@ -91,6 +87,19 @@ func (w *Writer) Write(rec *zed.Value) error {
 	return nil
 }
 
+func (w *Writer) append(val *zed.Value) {
+	n := len(w.vals)
+	if n >= cap(w.vals) {
+		newvals := make([]zed.Value, n, (n+1)*2)
+		for i, val := range w.vals {
+			newvals[i] = val
+		}
+		w.vals = newvals
+	}
+	w.vals = w.vals[:n+1]
```

Comment:
I think we should just use `append` in these situations. It calls `runtime.growslice`, which does 2x for small slices and 1.25x for large slices.
```suggestion
	if n < cap(w.vals) {
		w.vals = w.vals[:n+1]
	} else {
		w.vals = append(w.vals, zed.Value)
	}
```